### PR TITLE
e2e test for protocol upgrade

### PR DESCRIPTION
**Pull request by fixxxedpoint**
# Description

e2e-test that verifies that an attempt of protocol upgrade without a quorum of `interested` nodes should fail.

## Type of change

new e2e-test

### CODE REVIEW

File: bin/node/Cargo.toml [hunk 1]
```diff
@@ -84,3 +84,6 @@ try-runtime = [
 enable_treasury_proposals = [
     "aleph-runtime/enable_treasury_proposals"
 ]
+only_legacy = [
+    "finality-aleph/only_legacy"
+]
```
Comment (obrok) [inline bin/node/Cargo.toml:89]: Suggestion: add a newline.

File: finality-aleph/Cargo.toml [hunk 2]
```diff
@@ -59,3 +59,6 @@ sp-io = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch
 substrate-test-runtime-client = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.28" }
 substrate-test-runtime = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.28" }
 sc-block-builder = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.28" }
+
+[features]
+only_legacy = []
```
Comment (obrok) [inline finality-aleph/Cargo.toml:64]: Suggestion: add a newline.

**Review by obrok — APPROVED** [review]

File: e2e-tests/src/test/version_upgrade.rs [hunk 3]
```diff
@@ -36,3 +36,46 @@ pub fn schedule_version_change(config: &Config) -> anyhow::Result<()> {
 
     Ok(())
 }
+
+// A test that schedules a version upgrade that supposed to fail, awaits it, and checks if finalization stopped.
```
Comment (maciejzelaszczyk) [inline e2e-tests/src/test/version_upgrade.rs:40]: Nit: _that is supposed_

File: e2e-tests/src/test/version_upgrade.rs [hunk 4]
```diff
@@ -36,3 +36,46 @@ pub fn schedule_version_change(config: &Config) -> anyhow::Result<()> {
...
+    wait_for_at_least_session(&connection, session_after_upgrade)?;
+    let connection = connection.as_connection();
+    let finalized_block_head = connection.as_connection().get_finalized_head()?;
+    let finalized_block = connection.get_header::<Header>(finalized_block_head)?;
+
+    let finalized_block = match finalized_block {
+        Some(block) => block.number,
+        None => {
+            return Err(anyhow::Error::msg(
+                "somehow no block was finalized (even we saw one)",
```
Comment (maciejzelaszczyk) [inline e2e-tests/src/test/version_upgrade.rs:72]: Nit: _even though we_?

**Review by maciejzelaszczyk — APPROVED** [review]
LGTM

**Pull request merged**